share/man/man9/cdefs.9

```diff
@@ -0,0 +1,297 @@
+.\"-
+.\" Copyright (c) 2024 M. Warner Losh <imp@FreeBSD.org>
+.\"
+.\" SPDX-License-Identifier: BSD-2-Clause
+.\"
+.Dd July 3, 2024
+.Dt CDEFS 9
+.Os
+.Sh NAME
+.Nm cdefs
+.Nd "Compiler portability definitions"
+.Sh DESCRIPTION
+.In sys/cdefs.h
+defines macros for compiler, C language standard portability, POSIX standards
+compliance and symbol visibility.
+It defines programming interfaces for the system header files to adopt to the
+many environments
+.Fx
+supports compilation for.
+It defines convenience macros for the
+.Fx
+sources, tailored to the base
+system's needs.
+.Pp
+Most of these macros are for use inside the
+.Fx
+sources only.
+They are not intended as a general portability layer.
+.Sh Supported Compilers
+There are two levels of compiler support for
+.Fx :
+Compilers supported for building programs on
+.Fx .
+These are compilers that you can use to build and run programs on
+Fx .
+.Bl -column -offset indent ".Sy Compiler" ".Sy Versions"
```

Comment:
```suggestion
.Bl -column -offset indent "Compiler" "Versions"
```

What I was saying is you don't want to use macros in a width or offset argument. It behaves unpredictably in different implementations. In this case, on our default console stack, what you're doing actually looks great, but that's kinda lucky.

So, I left my laptop somewhere, but basically I played with all your tables and found something that seemed like the best compromise.

If you are going to use macros in a width or offset (in a Bl or Bd) then definitely omit the dot, because there can only be exactly one dot at the beginning or end of a line in roff. The dot lots of times seemingly expands to negative spaces, like it was doing in fwget.8

Reply:
I haven't found good language to document this unambiguously. E.g. sometimes you want to . That is reccomended use of a macro for the width. But `.Bl -tag -width ".Fl O"` does the opposite of what you want.

Reply:
OK. I copied this from somewhere (maybe mdoc man page). I'll remove them though, it's not at all important and I didn't do it for any reason beyond this.